### PR TITLE
EVM Fungible App

### PR DESCRIPTION
**Pull request by vovac12**

### CODE REVIEW

File: pallets/evm-fungible-app/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,839 @@
+//! # ERC20
```
Comment (N1ghtStorm) [inline pallets/evm-fungible-app/src/lib.rs]: Sora Licence headers are missing in all files

**Review by N1ghtStorm — APPROVED** [review]

File: pallets/evm-fungible-app/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,839 @@
...
+//! this event and relay it to the other chain.
+//!
+//! ## Interface
+//!
+//! ### Dispatchable Calls
+//!
+//! - `burn`: Burn an ERC20 token balance.
+#![cfg_attr(not(feature = "std"), no_std)]
+// TODO #167: fix clippy warnings
+#![allow(clippy::all)]
```
Comment (N1ghtStorm) [inline pallets/evm-fungible-app/src/lib.rs]: check clippy in all files

File: pallets/evm-fungible-app/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,839 @@
...
+            );
+            ensure!(
+                !TokenAddresses::<T>::contains_key(network_id, &asset_id),
+                Error::<T>::TokenAlreadyRegistered
+            );
+            TokenAddresses::<T>::insert(network_id, &asset_id, contract);
+            AssetsByAddresses::<T>::insert(network_id, contract, &asset_id);
+            AssetKinds::<T>::insert(network_id, &asset_id, asset_kind);
+            SidechainPrecision::<T>::insert(network_id, &asset_id, sidechain_precision);
+            T::AssetRegistry::manage_asset(network_id.into(), asset_id)?;
```
Comment (N1ghtStorm) [inline pallets/evm-fungible-app/src/lib.rs]: May we need a special event for registering asset?

File: pallets/evm-fungible-app/src/lib.rs (hunk 2 above)
Comment (vovac12) [reply pallets/evm-fungible-app/src/lib.rs]: Fixed

File: pallets/evm-fungible-app/src/lib.rs (hunk 3 above)
Comment (vovac12) [reply pallets/evm-fungible-app/src/lib.rs]: Added

File: pallets/evm-fungible-app/src/lib.rs (hunk 1 above)
Comment (vovac12) [reply pallets/evm-fungible-app/src/lib.rs]: Added

**Review by N1ghtStorm — APPROVED** [review]

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Pull request merged**